### PR TITLE
Add osx to travis-ci

### PR DESCRIPTION
**Pull request by jwillemsen**
* .travis.yml:

### CODE REVIEW

**Comment by jwillemsen** [comment]
Looks the warnings for the usage of register are causing travis-ci to fail for osx. travis is using ACE+TAO latest micro, fixes for that are on master

**Comment by mitza-oci** [comment]
The Travis config would need to be updated with -Wno-deprecated-register until after the DOC Group release.

**Comment by jwillemsen** [comment]
Should we do this in the travis-ci file or within configure? Probably configure is a better place given the possibility to also use OCI 2.2a which does use register

**Comment by mitza-oci** [comment]
It's only a problem for Travis, a normal build doesn't error-out due to this:

> The log length has exceeded the limit of 4 MB (this usually means that the test suite is raising the same exception over and over).
The job has been terminated

**Comment by jwillemsen** [comment]
I disabled inlining, that should also reduce the size of the log file

File: .travis.yml [hunk 1]
```diff
@@ -41,7 +41,9 @@ matrix:
           packages:
             - libxerces-c-dev
             - clang++-3.9
-      env: COMPILER=clang++-3.9
+      env: COMPILER=clang++
+    - os: osx
+      env: COMPILER=clang++ XERCES3=0 INLINE=0
```
Comment (mitza-oci) [inline .travis.yml]: There is already NO_INLINE=1 on line 10, let's not have both variables.

Comment (jwillemsen) [inline .travis.yml]: Missed that, updated the file

**Pull request merged**